data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultCosmosRepositoryOperations.java

```diff
@@ -246,27 +251,111 @@ public <R> Page<R> findPage(PagedQuery<R> query) {
     public <T> T persist(InsertOperation<T> operation) {
         CosmosContainer container = getContainer(operation);
         T entity = operation.getEntity();
-        ObjectNode tree = serializeToTree(entity, Argument.of(operation.getRootEntity()));
+        ObjectNode tree = serialize(entity, Argument.of(operation.getRootEntity()));
         container.createItem(tree, new CosmosItemRequestOptions());
         return entity;
     }
 
     @Override
     public <T> T update(UpdateOperation<T> operation) {
-        return null;
+        RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(operation.getRootEntity());
+        T entity = operation.getEntity();
+        CosmosContainer container = getContainer(persistentEntity);
+        ObjectNode item = serialize(entity, Argument.of(operation.getRootEntity()));
+        container.upsertItem(item);
```

Comment:
Well, in our case it is actually replace because upsert would create new document if does not exist. I have changed it to replace as it makes more sense.

Reply:
Imho upsert shouldn’t be a default, you can endup inserting something that you intended to update

Reply:
Yes, I agree and I changed it to replaceItem, tnx.

Reply:
Yes, again one use case if partition key is not defined. Method replaceItem requires partition key and id. Leaving it as is for now, but later I think we should make partition key to be required for a container (via settings or field annotation) and then we won't have issues with that. Anyway every user will need to defined partition key on a container.